port/webhook/resource.go

```diff
@@ -0,0 +1,143 @@
+var _ resource.Resource = &WebhookResource{}
+var _ resource.ResourceWithImportState = &WebhookResource{}
+
+func NewWebhookResource() resource.Resource {
+	return &WebhookResource{}
+}
+
+type WebhookResource struct {
+	portClient *cli.PortClient
+}
+
+func (r *WebhookResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_webhook"
+}
+
+func (r *WebhookResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.portClient = req.ProviderData.(*cli.PortClient)
+}
+
+func (r *WebhookResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state *WebhookModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	identifier := state.Identifier.ValueString()
+	w, statusCode, err := r.portClient.ReadWebhook(ctx, identifier)
+	if err != nil {
+		if statusCode == 404 {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("failed to read webhook", err.Error())
+		return
+	}
+
+	err = refreshWebhookState(ctx, state, w)
+	if err != nil {
+		resp.Diagnostics.AddError("failed writing webhook fields to resource", err.Error())
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *WebhookResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state *WebhookModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	w, err := webhookResourceToBody(ctx, state)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to convert webhook resource to body", err.Error())
+		return
+	}
+
+	wp, err := r.portClient.CreateWebhook(ctx, w)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create webhook", err.Error())
+		return
+	}
+
+	writeWebhookComputedFieldsToState(state, wp)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func writeWebhookComputedFieldsToState(state *WebhookModel, wp *cli.Webhook) {
+	state.ID = types.StringValue(wp.Identifier)
+	state.Identifier = types.StringValue(wp.Identifier)
+	state.CreatedAt = types.StringValue(wp.CreatedAt.String())
+	state.CreatedBy = types.StringValue(wp.CreatedBy)
+	state.UpdatedAt = types.StringValue(wp.UpdatedAt.String())
+	state.UpdatedBy = types.StringValue(wp.UpdatedBy)
+}
+
+func (r *WebhookResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state *WebhookModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	w, err := webhookResourceToBody(ctx, state)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to convert webhook resource to body", err.Error())
+		return
+	}
```

Comment:
you put this after every `webhookResourceToBody` call

maybe just put the error handling inside?